spec/isolation/components/model/with_hanami_model_spec.rb

```diff
@@ -10,5 +10,38 @@
         expect(Hanami::Model.configuration.logger).to eq(Hanami.logger)
       end
     end
+
+    it "disconnects database connections on reboot" do
+      with_project do
+        require Pathname.new(Dir.pwd).join("config", "environment")
+
+        # Simulate previous connection
+        Hanami::Components.resolve('model.configuration')
+        Hanami::Components.resolve('model.sql')
+        Hanami::Model.load!
+
+        #
+        Hanami::Model.configuration.connection.test_connection
+        # Current connections count is 1
+
+        #
+        Hanami::Components.resolve('model')
+        # Current connections count is 0
+
+        expect(Hanami::Components['model']).to     be(true)
+        expect(Hanami::Components['model.sql']).to be(true)
+        expect(Hanami::Model.configuration.logger).to eq(Hanami.logger)
+
+        # This is tight coupled to Sequel
+        #
+        # When `.disconnect` is invoked, it returns a collection of disconnected
+        # connections. Here we want to assert that `.disconnect` was invoked as
```

Comment:
I think that he's testing if `disconnect` is doing it's thing and not it it's being called (as he's calling it explicitly)